File: bsep-frontend/src/types/passwordEntry.ts
export interface PasswordEntry {
  id: number;
  website: string;
  username: string;
  encryptedPassword: string;
  certificateSerialNumber: string;
  certificateCommonName: string;
  createdAt: string;
}

export interface CreatePasswordEntryRequest {
  website: string;
  username: string;
  password: string;
  certificateSerialNumber: string;
}

export interface DecryptedPasswordEntry {
  id: number;
  website: string;
  username: string;
  decryptedPassword: string;
}

File: bsep_backend/src/main/java/com/example/bsep_backend/passwordmanager/domain/PasswordEntry.java
package com.example.bsep_backend.passwordmanager.domain;

import com.example.bsep_backend.domain.User;
import com.example.bsep_backend.pki.domain.Certificate;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "password_entries")
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PasswordEntry {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String website;

    @Column(nullable = false)
    private String username;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String encryptedPassword;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "certificate_id", nullable = false)
    private Certificate certificate;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();
}
